### PR TITLE
update icewind/smb to 1.0.4

### PR DESCRIPTION
**Pull request by icewind1991**
Mostly includes workspace/domain related fixes and a fix relating to deleting.

Note that I've not been able to fully test the workspace/domain logic myself because I cant seem to manage to reproduce the issues some people are having.

### CODE REVIEW

File: apps/files_external/3rdparty/icewind/smb/.travis.yml [hunk 1]
```diff
@@ -3,21 +3,29 @@ php:
   - 5.3
   - 5.4
   - 5.5
+  - 5.6
```
Comment (DeepDiver1975) [inline apps/files_external/3rdparty/icewind/smb/.travis.yml]: add to .gitignore as well?

File: apps/files_external/3rdparty/icewind/smb/install_libsmbclient.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
```
Comment (DeepDiver1975) [inline apps/files_external/3rdparty/icewind/smb/install_libsmbclient.sh]: add to .gitignore as well?

File: apps/files_external/3rdparty/icewind/smb/.travis.yml (hunk 1 above)
Comment (icewind1991) [inline apps/files_external/3rdparty/icewind/smb/.travis.yml]: done

**Comment by Xenopathic** [comment]
:+1:

**Comment by icewind1991** [comment]
Added the update to 1.0.4 which has improved handling of special characters

cc @DeepDiver1975 @PVince81

**Comment by DeepDiver1975** [comment]
Can I see the result of autotest-external on this one? THX

**Comment by icewind1991** [comment]
```
Waiting 5 seconds for smbd initialization ... 
PHPUnit 4.8.4 by Sebastian Bergmann and contributors.

Runtime:        PHP 5.6.12
Configuration:  /srv/http/owncloud/tests/phpunit-autotest-external.xml
Warning:        Deprecated configuration setting "strict" used
Warning:        The Xdebug extension is not loaded
                No code coverage will be generated.

................................................................. 65 / 70 ( 92%)
.....

Time: 786 ms, Memory: 22.50Mb

OK (70 tests, 262 assertions)

```

```
start: start-smb-windows.sh
name: smb-windows
PHPUnit 4.8.4 by Sebastian Bergmann and contributors.

Runtime:        PHP 5.6.12
Configuration:  /srv/http/owncloud/tests/phpunit-autotest-external.xml
Warning:        Deprecated configuration setting "strict" used
Warning:        The Xdebug extension is not loaded
                No code coverage will be generated.

................................................................. 65 / 70 ( 92%)
.....

Time: 3.29 seconds, Memory: 22.50Mb

OK (70 tests, 262 assertions)
```

Tested with both smbclient and libsmbclient backends

**Comment by PVince81** [comment]
Seems SMB still works :+1:

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by icewind1991** [comment]
Do we want to backport to 8.1?

**Comment by DeepDiver1975** [comment]
> Do we want to backport to 8.1?

yes please

**Comment by DeepDiver1975** [comment]
@karlitschek ^ FYI

**Comment by karlitschek** [comment]
as discussed. please backport! :+1:

**Comment by icewind1991** [comment]
8.1: 18397

**Pull request merged**